gum/meson.build

```diff
@@ -289,7 +300,7 @@ if have_gir
 
   gnome.generate_gir(
     gum,
-    sources: gum_sources + gum_headers,
+    sources: gum_sources + gum_headers + [gum_enum_h],
```

Comment:
I think we have some handwritten _get_type() implementations for some of our enums – won't this conflict? If so we will also need to update the Windows build system to use a custom build step. (I can assist with this part if you don't have access to a Windows dev env.)

Reply:
Yes. We need to root these out and change them so they are properly formatted enums; is this appropriate for this PR? `GumEventType` being a prime example. And yes, I will need to enlist your help for ensuring this PR works on Windows :smile:.

Reply:
Cool! Yep, I think we should deal with it in this PR.

I think it's only these three:
```sh
% git grep '_get_type' gum | grep "\.c" | grep "(void)"
gum/gum.c:gum_cpu_type_get_type (void)
gum/gumdarwinmodule.c:gum_darwin_module_flags_get_type (void)
gum/gumprocess.c:gum_code_signing_policy_get_type (void)
%
```

By “properly formatted enums”, do you mean that the `get_type()` is generated, or also that e.g. `GUM_CALL` needs to become `GUM_EVENT_TYPE_CALL`? (Haven't looked into what GIR expects, but I imagine it can be overridden through annotations.)

Reply:
`GUM_CALL` needs to be `GUM_EVENT_TYPE_CALL`, yeah. I haven't looked into other enums in the Gum codebase yet, so perhaps you have more insight on that. Regardless, I think we should fix the `get_type()`s in this PR, and then deal with the malformed enum names in another?

Reply:
Ok. Well, there's got to be an annotation to override this, as I recall e.g. GStreamer has several such enums, for instance [GstFlowReturn](https://gstreamer.freedesktop.org/documentation/gstreamer/gstpad.html?gi-language=c#GstFlowReturn) with members such as `GST_FLOW_OK`.

Reply:
I was wrong: we are not `typedef enum`ing, which is what is causing this problem. So, we'll update all enums to use this format (ABI breaking change, since we currently use `guint` for all enum types).

Reply:
For review purposes, here are the generated files for these enums (and the others): https://gist.github.com/meme/b7eb683a6294c04469bf95cc791290e1